Version 1.7 broke the USB support for F103C8

I upgraded to 1.7 from 1.6.1 and the USB on all BluePills that I have stopped working. Switched back to 1.6.1 and everything works. 1.7 is basically unusable on the most used maker's board now. No code is changed in the sketches - just recompiled with 1.7.

It works again - Thank you ! What does that function do / change ? 

Btw can you put "__attribute__ ((unused))" before those params coz it's annoying and fills the screen and I really want -Wall parameter - it saved me few times  :) 

In file included from C:\Users\tbyte\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.7.0\cores\arduino\stm32\HAL\stm32yyxx_hal_tim_ex.c:5:
C:\Users\tbyte\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.7.0\system/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_tim_ex.c: In function 'HAL_TIMEx_RemapConfig':
C:\Users\tbyte\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.7.0\system/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_tim_ex.c:1734:60: warning: unused parameter 'htim' [-Wunused-parameter]
 HAL_StatusTypeDef HAL_TIMEx_RemapConfig(TIM_HandleTypeDef *htim, uint32_t Remap)
                                         ~~~~~~~~~~~~~~~~~~~^~~~
C:\Users\tbyte\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.7.0\system/Drivers/STM32F1xx_HAL_Driver/Src/stm32f1xx_hal_tim_ex.c:1734:75: warning: unused parameter 'Remap' [-Wunused-parameter]
 HAL_StatusTypeDef HAL_TIMEx_RemapConfig(TIM_HandleTypeDef *htim, uint32_t Remap)
                                                                  ~~~~~~~~~^~~~~
In file included from C:\Users\tbyte\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.7.0\cores\arduino\stm32\usb\usb_device_ctlreq.c:3:
C:\Users\tbyte\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.7.0\system/Middlewares/ST/STM32_USB_Device_Library/Core/Src/usbd_ctlreq.c: In function 'USBD_CtlError':
C:\Users\tbyte\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.7.0\system/Middlewares/ST/STM32_USB_Device_Library/Core/Src/usbd_ctlreq.c:853:42: warning: unused parameter 'req' [-Wunused-parameter]
                    USBD_SetupReqTypedef *req)
                    ~~~~~~~~~~~~~~~~~~~~~~^~~

--- a/system/Middlewares/ST/STM32_USB_Device_Library/Core/Src/usbd_core.c
+++ b/system/Middlewares/ST/STM32_USB_Device_Library/Core/Src/usbd_core.c
@@ -393,12 +393,14 @@ USBD_StatusTypeDef USBD_LL_DataInStage(USBD_HandleTypeDef *pdev,
         }
         else
         {
-          if ((pdev->pClass->EP0_TxSent != NULL) &&
-              (pdev->dev_state == USBD_STATE_CONFIGURED))
+          if (pdev->dev_state == USBD_STATE_CONFIGURED)
           {
-            pdev->pClass->EP0_TxSent(pdev);
+            if (pdev->pClass->EP0_TxSent != NULL)
+            {
+              pdev->pClass->EP0_TxSent(pdev);
+            }
+            USBD_LL_StallEP(pdev, 0x80U);
           }
-          USBD_LL_StallEP(pdev, 0x80U);
           USBD_CtlReceiveStatus(pdev);
         }
       }

